packages/govern/internal/actions/lib/AbstractAction.ts

```diff
@@ -0,0 +1,72 @@
+import { DocumentNode } from 'graphql'
+import Configuration from '../../configuration/Configuration'
+
+/**
+ * @abstract AbstractAction
+ */
+export default abstract class AbstractAction {
+  /**
+   * Contains the GraphQL query of the current action
+   *
+   * @var {string} gqlQuery
+   *
+   * @protected
+   */
+  protected gqlQuery: DocumentNode
+
+  /**
+   * The current configuration to execute this action
+   *
+   * @var {Configuration} configuration
+   *
+   * @protected
+   */
+  protected configuration: Configuration
+
+  /**
+   * The required parameters for this action
+   *
+   * @param {any} parameters
+   *
+   * @protected
+   */
+  protected parameters?: any
+
+  /**
+   * @param {any} parameters - The required parameters for this action
+   *
+   * @constructor
+   */
+  constructor(parameters?: any) {
+    this.configuration = Configuration.get()
+    this.parameters = this.validateParameters(parameters)
+  }
+
+  /**
+   * Validates the given parameters
+   *
+   * @method validateParameters
+   *
+   * @param {any} parameters
+   *
+   * @returns {any}
+   */
+  protected validateParameters(parameters?: any): any {
+    return parameters;
```

Comment:
We might want to run prettier on all the files:

```suggestion
    return parameters
```